### PR TITLE
Add groups show

### PR DESCRIPTION
**Pull request by jsteiner**

### CODE REVIEW

File: lib/cog_api/client.ex [hunk 1]
```diff
@@ -6,15 +6,16 @@ defmodule CogApi.Client do
...
   @callback role_create(%Endpoint{}, %{}) :: {atom, %Role{}}
   @callback role_update(%Endpoint{}, String.t, %{}) :: {atom, %Role{}}
   @callback role_delete(%Endpoint{}, String.t) :: atom
+
+  @callback permission_index(%Endpoint{}) :: {atom, [%Permission{}]}
+  @callback permission_create(%Endpoint{}, String.t) :: {atom, %Permission{}}
+
+  @callback group_index(%Endpoint{}) :: {atom, [%Group{}]}
+  @callback group_show(%Endpoint{}, String.t) :: {atom, %Group{}}
+  @callback group_create(%Endpoint{}, %{}) :: {atom, %Group{}}
```
Comment (paulcsmith) [inline lib/cog_api/client.ex]: Why were these moved below? It seems like they were sorted alphabetically before

File: test/unit/cog_api/fake/groups_test.exs [hunk 2]
```diff
@@ -18,6 +18,19 @@ defmodule CogApi.Fake.GroupsTest do
     end
   end
 
+  describe "group_show" do
+    context "when the group exists" do
```
Comment (paulcsmith) [inline test/unit/cog_api/fake/groups_test.exs]: Do we care about the case when the group does not exist as well?

**Comment by paulcsmith** [comment]
Just a couple questions, but the it looks good to me

File: lib/cog_api/client.ex (hunk 1 above)
Comment (jsteiner) [inline lib/cog_api/client.ex]: The two files that depend on this were sorted the other way. Decided to change one instead of the other two. I'm going to put it back, then fix the other two in another commit.

File: test/unit/cog_api/fake/groups_test.exs (hunk 2 above)
Comment (jsteiner) [inline test/unit/cog_api/fake/groups_test.exs]: Not yet
